### PR TITLE
fix duplicated routes when reimporting jdl

### LINKED ISSUE #9178: Routes in entity.module.ts are duplicated upon reimporting jdl
##### **Overview of the issue**

If you make changes to a jdl file, and re-generate the model, the routes in entity.module.ts are duplicated (the generator fails to see they already exist).

##### **Motivation for or Use Case**

Route shouldn't be duplicated

##### **Reproduce the error**

1. Create a JDL with an entity
1. Import it
1. Modify a field in the JDL
1. Re-import it
1. See entity.module.ts : the entity's route is present twice

##### **Suggest a Fix**

I have a fix in preparation

##### **JHipster Version(s)**

5.8.1

##### **Browsers and Operating System**

OS X 10.14

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by clement26695**
Fix #9178 

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@afillatre : first of all, sorry for taking too much time for https://github.com/jhipster/generator-jhipster/issues/9178

The last time I tried your PR https://github.com/jhipster/generator-jhipster/pull/9179 it didn't fix the issue.
I just tested this PR and it works well. Maybe I don't test all use cases or I miss something ?
Does it look good for you ?

@wmarques @avdev4j : can you test it plz ?

I'd like to merge this to close this old ticket, but feel free to test / comment / review before.

**Comment by pascalgrimaud** [comment]
cc @deepu105 : I know you are busy but if you have time to have a quick look and review

**Comment by avdev4j** [comment]
This solution is close that the one I suggested intitially.
Even if it works I think it does not fix the real issue about the comma, prettier and file manipulation. After some weeks we didn't find a better way to satisfy everyone.

Except if we have a magic solution I think this is good enough to be used.

Thanks @ clement26695

File: generators/client/needle-api/needle-client-angular.js [hunk 1]
```diff
@@ -180,29 +180,38 @@ module.exports = class extends needleClientBase {
...
-                'jhipster-needle-add-entity-route',
-                this.generator.stripMargin(splicable)
+                `path: '${entityUrl}'`,
+                this.generator
             );
 
-            this.addBlockContentToFile(rewriteFileModel, errorMessage);
+            if (!isSpecificEntityAlreadyGenerated) {
+                const appName = this.generator.getAngularXAppName();
+                const isEntityAlreadyGenerated = jhipsterUtils.checkStringInFile(entityModulePath, 'loadChildren', this.generator);
```
Comment (avdev4j) [inline generators/client/needle-api/needle-client-angular.js]: Maybe a variable name like "isAnyEntityAlreadyGenerated" would be more clearer

**Comment by avdev4j** [comment]
I've just test a full import twice and it works. Routes are not duplicated anymore.

**Comment by pascalgrimaud** [comment]
@clement26695 : can you fix the prettier formatting plz ?

**Comment by pascalgrimaud** [comment]
Forget my comment, just pushed to your branch :)

**Pull request merged**